pkg/statsd/receiver.go

```diff
@@ -65,10 +71,15 @@ func (mr *MetricReceiver) RunMetrics(ctx context.Context) {
 // Receive accepts incoming datagrams on c, parses them and calls Handler.DispatchMetric() for each metric
 // and Handler.DispatchEvent() for each event.
 func (mr *MetricReceiver) Receive(ctx context.Context, c net.PacketConn) {
-	buf := make([]byte, packetSizeUDP)
+	conn := ipv6.NewPacketConn(c)
+	messages := make([]ipv6.Message, receiveBatchSize)
```

Comment:
You don't need to allocate all of that on each `Receive()` invocation. Make it a field of `MetricReceiver`.

Reply:
Receive is invoked once per goroutine, the buffers belong to the local goroutine, not the MR.

Reply:
Ok, then we need to do something about it to avoid reallocation. That is my point :)

Reply:
What reallocation?  It's allocated once outside of the forever loop?

Reply:
Yeah this memory shouldn't shared across goroutines. These buffers are (potentially) used for every invocation of ReadBatch. Each Receive invocation needs it's own set of buffers.

Reply:
I misread the code, ignore this comment, sorry.